classes/announcement/Repository.php

```diff
@@ -161,4 +196,132 @@ public function deleteMany(Collector $collector)
             $this->delete($announcement);
         }
     }
+
+    /**
+     * The subdirectory where announcement images are stored
+     */
+    public function getImageSubdirectory(): string
+    {
+        return 'announcements';
+    }
+
+    /**
+     * Get the base URL for announcement file uploads
+     */
+    public function getFileUploadBaseUrl(?Context $context = null): string
+    {
+        return join('/', [
+            Application::get()->getRequest()->getPublicFilesUrl($context),
+            $this->getImageSubdirectory(),
+        ]);
+    }
+
+    /**
+     * Handle image uploads
+     *
+     * @throws StoreTemporaryFileException Unable to store temporary file upload
+     */
+    protected function handleImageUpload(Announcement $announcement): void
+    {
+        $image = $announcement->getImage();
+        if ($image && $image['temporaryFileId']) {
+            $user = Application::get()->getRequest()->getUser();
+            $image = $announcement->getImage();
+            $temporaryFileManager = new TemporaryFileManager();
+            $temporaryFile = $temporaryFileManager->getFile((int) $image['temporaryFileId'], $user?->getId());
```

Comment:
No big deal, but I'd suggest not using the nullsafe operator on `$user?->getId()`. If somehow we make it here without a `$user`, a big fat fatal error is definitely a good thing.